moirae/models/ecm/__init__.py

```diff
@@ -46,53 +46,66 @@ def update_transient_state(
         # hyst_(k+1) = [see code, it's messy]
 
         # Get basic info
-        delta_t = new_inputs.time - previous_inputs.time
-        current_k = previous_inputs.current
-        temp_k = previous_inputs.temperature
-        current_kp1 = new_inputs.current
-        current_slope = 0.0 if self.current_behavior == 'constant' else (current_kp1 - current_k) / delta_t
+        delta_t = new_inputs.time - previous_inputs.time  # shape (time_batch, 1)
+        current_k = previous_inputs.current  # shape (current_batch, 1)
+        temp_k = previous_inputs.temperature  # shape (temp_batch, 1)
+        current_kp1 = new_inputs.current  # shape (current_batch, 1)
+        current_slope = np.atleast_2d(0.0) if self.current_behavior == 'constant' \
+            else (current_kp1 - current_k) / delta_t  # shape (trans_batch, 1)
 
         # We will assume that all health parameters remain constant between time
         # steps, independent of temperature or SOC variations. The value used
         # will be the one at the previous SOC and temperature values.
 
         # Set Coulombic efficiency to 1. if discharging
-        coul_eff = 1 if current_k < 0 else asoh.ce
+        coul_eff = np.atleast_2d(1) if current_k < 0 else asoh.ce  # shape (ce_batch, 1)
```

Comment:
Not a required change, but a numpy suggestion: consider `np.ones((1, 1))` or `np.ones_line(asoh.ce)` instead.

Reply:
Is there a reason for choosing one or the other among the three options?